apis/meta/v1/types.go

```diff
@@ -37,4 +37,9 @@ type ServiceAccountSelector struct {
 	// to the namespace of the referent.
 	// +optional
 	Namespace *string `json:"namespace,omitempty"`
+	// Audience specifies the `aud` claim for the service account token
+	// If the service account uses a well-known annotation for e.g. IRSA or GCP Workload Identity
+	// then this audiences will be appended to the list
+	// +optional
+	Audiences []string `json:"audiences,omitempty"`
```

Comment:
The idea is to add that `audiences` everywhere. provider can use/set a default which can be extended with this field.
It has been implemented for all providers.